libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -70,21 +80,17 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
             BotAssert.ActivityNotNull(activity);
             ClaimsIdentity claimsIdentity =  await JwtTokenValidation.AuthenticateRequest(activity, authHeader, _credentialProvider, _httpClient);
 
-            // For requests from channel App Id is in Audience claim of JWT token. For emulator it is in AppId claim. For 
-            // unauthenticated requests we have anonymouse identity provided auth is disabled.
-            string botAppId = GetBotId(claimsIdentity);
-            var context = new BotFrameworkTurnContext(botAppId, this, activity);
+            var context = new TurnContext(this, activity);
+            context.Services.Add<IIdentity>(claimsIdentity);
```

Comment:
Same comment as above. Registration of this seems unnecessary, but if you do it consider using the same explicit key approach mentioned above.

Reply:
wildo.